src/main/proto/com/google/api/codegen/samplegen/v1/sample_config_v1.proto

```diff
@@ -0,0 +1,246 @@
+  // request field. If this is not provided or empty, the `value` of
+  // this `field` is simply hard-coded inside the body of the sample
+  // function. Otherwise, this `field` can be specified as a parameter
+  // to the sample function and as a command-line argument (passed to
+  // the sample function) with default value `value`.
+  bool input_parameter = 5;
+}
+
+// A single statement used in handling an RPC response. The statement
+// must be exactly one of the defined statement fields; there is one
+// field per statement type.
+//
+// Some statements accept variable references (eg
+// `$resp.names[1].first`), which are defined as sequences of dotted
+// variable names and indexed variable references:
+//
+// - a variable name is a sequence of alphanumeric characters
+//   and/or underscores, beginning with an alphabetic character. All
+//   alphabetic characters are case-insensitive and will be correctly
+//   interpreted as protobuf fields when referring to those, and
+//   output, if needed, as properly formatted variables in each target
+//   language. The special variable name `$resp` refers to the
+//   response obtained from the RPC.
+//
+// - the variable name before the first separator character
+//   (dot, bracket, or brace) must refer to a variable within the
+//   current lexical scope: either `$resp`, denoting the RPC response,
+//   or a variable name previously defined in a surrounding
+//   `loop` or `define` statement.
+//
+// - dotted notation is used to refer to subfields square brackets
+//
+// - `[]` are used to index into a list. The index must be a
+//   literal number or variable reference resolving to a number
+//
+// - curly braces `{}` are used to key into a map. The index must
+//   be a literal or variable reference.
+message ResponseStatement {
+
+  // A statement that iterates over lists or maps.
+  message LoopStatement {
+    // A variable reference denoting the list (repeated field) over
+    // which to iterate.
+    //
+    // Exactly one of `map` and `collection` should be specified.
+    string collection = 1;
+
+    // A variable reference denoting the iteration variable.  This
+    // must be specified iff `collection` is specified.
+    string variable = 2;
+
+    // A variable reference to the map over which to iterate. At least
+    // one of the fields `key` or `value` must be specified, and will
+    // cause the loop to iterate over the map keys, the map values, or
+    // both keys and values.
+    //
+    // Exactly one of `map` and `collection` should be specified.
+    string map = 4;
+
+    // The iteration variable for key.
+    //
+    // At least one of `key` and `value` must be specified if `map` is
+    // specified.
+    string key = 5;
+
+    // The iteration variable for value.
+    //
+    // At least one of `key` and `value` must be specified if `map` is
+    // specified.
+    string value = 6;
+
+    // The contents of the loop.
+    repeated ResponseStatement body = 3;
```

Comment:
I'd number these sequentially---not that it matters.

(I realize I neglected to do this when I wrote this proto for the design doc)